server/middleware/authentication.js

```diff
@@ -98,12 +98,22 @@ export const _authenticateUserByJwt = async (req, res, next) => {
     return;
   }
 
+  if (req.jwtPayload.scope === 'twofactorauth') {
+    return next(errors.Unauthorized('Cannot use this token on this route.'));
+  }
+
   /**
    * Functionality for one-time login links. We check that the lastLoginAt
    * in the JWT matches the lastLoginAt in the db. If so, we allow the user
    * to log in, and update the lastLoginAt.
    */
   if (req.jwtPayload.scope === 'login') {
+    const path = req.path;
+
+    if (path !== '/users/update-token') {
```

Comment:
Thank you, not beautiful but doing the job. Can you add some code comments explaining and future steps for this?